src/components/SplitPane.tsx

```diff
@@ -133,6 +142,59 @@ export function SplitPane(props: SplitPaneProps) {
 
   const isFinalClosed = isParentClosed ? true : isSidePaneClosed;
 
+  const getSplitInnerPaneStyle = useCallback(
+    (side: SideSeparation) => {
+      return cssStyles.item(
+        isFinalClosed,
+        sideSeparation === side,
+        orientation,
+        size,
+        type,
+      );
+    },
+    [isFinalClosed, orientation, sideSeparation, size, type],
+  );
+
+  const resizeObserverHandler = useCallback(
+    ({ width, height }) => {
+      if (parentRef.current) {
+        const { width: parentWidth, height: parentHeight } =
+          parentRef.current.getBoundingClientRect();
+
+        const parentDimension = [parentWidth, parentHeight];
+        const panelDimension = [width, height];
+        const sizeIndex = orientation === 'horizontal' ? 0 : 1;
+        const size =
+          sideSeparation === 'start'
+            ? panelDimension[sizeIndex]
+            : parentDimension[sizeIndex] - panelDimension[sizeIndex];
```

Comment:
There is probably a bug somewhere else if you absolutely need use the size of different divs depending on the `sideSeparation`

Reply:
I assume I only track the change of the size for one side of the splitPane so if we change the sideSpeapartion === "end" then we need to call the size ``` parentDimension[sizeIndex] - panelDimension[sizeIndex]```.

but yes we can get rid of the parentRef  from the resizer handler function if we are going to pass it to the onResize call-back based on the sideSeparation

Reply:
@lpatiny What size should be tracked in your opinion? The total (content+separator+panel), the content, or the panel?

Reply:
Why not pass the resize observer's `ref` conditionally to the correct child?

Reply:
Right that does not work but I think you can follow Michael's suggestion to pass the ref to the one or the other based on `sideSeparation`